### PR TITLE
Add Docker Build Metadata Arguments

### PR DESCRIPTION
**Pull request by SaikrishnaBairamoni**
# PR Details
This pull request updates our reusable GitHub Actions workflows to enhance the handling and incorporation of build metadata (`BUILD_DATE`, `VCS_REF`, `VERSION`) into our Docker images. The updates adds a new directory docker-build-metadata with action.yml script which determines the build metadata.
## Related Jira Key
[ARC-80
](https://usdot-carma.atlassian.net/browse/ARC-80)

## Motivation and Context
Metadata in CI images is useful for debugging

## How Has This Been Tested?
CI/CD

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/dockerhub.yml [hunk 1]
```diff
@@ -68,17 +72,24 @@ jobs:
           else
             echo git_branch="$GITHUB_REF_NAME" >> $GITHUB_OUTPUT
           fi
-      - name: Build
+      - name: Determine Build Metadata
+        id: build-metadata
+        uses: usdot-fhwa-stol/actions/docker-build-metadata@feature-add-docker-build-metadata
```
Comment (paulbourelly999) [inline .github/workflows/dockerhub.yml:77]: this needs to be main

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
       aws-secret-access-key: ${{ secrets.APT_REPO_AWS_SECRET_ACCESS_KEY }}
   test-dockerhub:
     if: ${{ github.event_name }} != "pull_request"
-    uses: usdot-fhwa-stol/actions/.github/workflows/dockerhub.yml@main
+    uses: usdot-fhwa-stol/actions/.github/workflows/dockerhub.yml@feature-add-docker-build-metadata
```
Comment (paulbourelly999) [inline .github/workflows/main.yml:22]: this needs to be main

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name: Test STOL actions
 on:
   push:
     branches:
-      - main
+      - feature-add-docker-build-metadata
```
Comment (paulbourelly999) [inline .github/workflows/main.yml:6]: this needs to be main

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
Update to main

File: .github/workflows/dockerhub.yml (hunk 1 above)
Comment (SaikrishnaBairamoni) [reply .github/workflows/dockerhub.yml]: updated

File: .github/workflows/main.yml (hunk 2 above)
Comment (SaikrishnaBairamoni) [reply .github/workflows/main.yml]: updated

File: .github/workflows/main.yml (hunk 3 above)
Comment (SaikrishnaBairamoni) [reply .github/workflows/main.yml]: updated

**Review by paulbourelly999 — APPROVED** [review]

**Pull request merged**